kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -365,23 +365,21 @@ impl<'tcx> GotocCtx<'tcx> {
     fn codegen_switch_int(
         &mut self,
         discr: &Operand<'tcx>,
-        switch_ty: Ty<'tcx>,
         targets: &SwitchTargets,
         loc: Location,
     ) -> Stmt {
         let v = self.codegen_operand(discr);
-        let switch_ty = self.monomorphize(switch_ty);
+        let switch_ty = v.typ().clone();
```

Comment:
Not really. The `switch_ty` field was removed from the `SwitchInt` struct in https://github.com/rust-lang/rust/pull/105234 because it's redundant: its type is always the same as the type of the `discr` field. So my change just copies the type that we've already created for the `discr` field.

See this comment for more info: https://github.com/rust-lang/rust/pull/105234/files#diff-3a6077a453e9ef35b5b85b4419025066cc7a0f3bad1207ae98860902c817a252L531

Reply:
But yes, AFAIK, the monomorphization happens in `codegen_operand`.